# HG changeset patch
# User John Donoghue <john.donoghue@ieee.org>
# Date 1384741909 18000
#      Sun Nov 17 21:31:49 2013 -0500
# Node ID a4bc3ad4c703fedccd492fca0a7545d2b4f49dca
# Parent  265f9186e543aa02c05d52bbc2f4979d2d2ba2a4
Use gnulib uname

* configure.ac: remove check for uname and sys/utsname.h.

* liboctave/system/oct-uname.cc (octave_uname::init):
  remove #ifdef HAVE_SYS_UTSNAME_H from uname function.

* bootstrap.conf: add uname to gnulib_modules.

diff --git a/bootstrap.conf b/bootstrap.conf
--- a/bootstrap.conf
+++ b/bootstrap.conf
@@ -87,16 +87,17 @@ gnulib_modules="
   sys_stat
   sys_time
   sys_times
   time
   times
   tmpfile
   trunc
   truncf
+  uname
   unistd
   unlink
   vasprintf
 "
 
 # Additional xgettext options to use.  Use "\\\newline" to break lines.
 XGETTEXT_OPTIONS=$XGETTEXT_OPTIONS'\\\
  --from-code=UTF-8\\\
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2019,17 +2019,17 @@ AC_HEADER_SYS_WAIT
 
 ## C headers
 
 dnl Use multiple AC_CHECKs to avoid line continuations '\' in list
 AC_CHECK_HEADERS([curses.h direct.h dlfcn.h floatingpoint.h fpu_control.h])
 AC_CHECK_HEADERS([grp.h ieeefp.h inttypes.h locale.h memory.h ncurses.h])
 AC_CHECK_HEADERS([poll.h pthread.h pwd.h sunmath.h sys/ioctl.h])
 AC_CHECK_HEADERS([sys/param.h sys/poll.h sys/resource.h])
-AC_CHECK_HEADERS([sys/select.h sys/utsname.h termcap.h])
+AC_CHECK_HEADERS([sys/select.h termcap.h])
 
 ## C++ headers
 
 AC_LANG_PUSH(C++)
 
 AC_CHECK_HEADERS([sstream])
 OCTAVE_UNORDERED_MAP_HEADERS
 
@@ -2140,17 +2140,17 @@ AC_CHECK_FUNCS([endgrent endpwent execvp
 AC_CHECK_FUNCS([getegid geteuid getgid getgrent getgrgid getgrnam])
 AC_CHECK_FUNCS([getpgrp getpid getppid getpwent getpwuid getuid])
 AC_CHECK_FUNCS([isascii kill])
 AC_CHECK_FUNCS([lgamma lgammaf lgamma_r lgammaf_r])
 AC_CHECK_FUNCS([log1p log1pf pipe])
 AC_CHECK_FUNCS([realpath resolvepath roundl])
 AC_CHECK_FUNCS([select setgrent setpwent siglongjmp strsignal])
 AC_CHECK_FUNCS([tempnam tgammaf toascii])
-AC_CHECK_FUNCS([umask uname waitpid])
+AC_CHECK_FUNCS([umask waitpid])
 AC_CHECK_FUNCS([_kbhit])
 
 dnl There are no workarounds in the code for missing these functions.
 AC_CHECK_FUNCS([modf pow sqrt sqrtf], [],
                [AC_MSG_ERROR([Missing function required to build Octave])])
 
 ## exp2, round, tgamma function checks
 AC_LANG_PUSH(C++)
diff --git a/liboctave/system/oct-uname.cc b/liboctave/system/oct-uname.cc
--- a/liboctave/system/oct-uname.cc
+++ b/liboctave/system/oct-uname.cc
@@ -22,34 +22,30 @@ along with Octave; see the file COPYING.
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <cerrno>
 #include <cstring>
 
-#ifdef HAVE_SYS_UTSNAME_H
 #include <sys/utsname.h>
-#endif
 
 #include "oct-uname.h"
 
 void
 octave_uname::init (void)
 {
-#if defined (HAVE_UNAME) && defined (HAVE_SYS_UTSNAME_H)
   struct utsname unm;
 
   err = ::uname (&unm);
 
   if (err < 0)
     msg = gnulib::strerror (errno);
   else
     {
       utsname_sysname = unm.sysname;
       utsname_nodename = unm.nodename;
       utsname_release = unm.release;
       utsname_version = unm.version;
       utsname_machine = unm.machine;
     }
-#endif
 }
